domain_models/views.py

```diff
@@ -0,0 +1,170 @@
+
+        if not issubclass(model_cls, models.DomainModel):
+            raise TypeError("Attribute __model_cls__ must be subclass of "
+                            "DomainModel.")
+
+    @classmethod
+    def chk_incl_excl(mcs):
+        """Check __include__ and __exclude__ attributes."""
+        include = mcs.attributes.get('__include__', tuple())
+        exclude = mcs.attributes.get('__exclude__', tuple())
+
+        if not isinstance(include, tuple):
+            raise TypeError("Attribute __include__ must be a tuple.")
+
+        if not isinstance(exclude, tuple):
+            raise TypeError("Attribute __exclude__ must be a tuple.")
+
+        if all((not include, not exclude)):
+            return None
+
+        if all((include, exclude)):
+            raise AttributeError("Usage of __include__ and __exclude__ "
+                                 "at the same time is prohibited.")
+
+        include_names = [item.name for item in include]
+        exclude_names = [item.name for item in exclude]
+
+        mcs.chk_intersections(include_names, exclude_names)
+        mcs.match_unknown_attrs(include_names, exclude_names)
+
+    @classmethod
+    def chk_intersections(mcs, include, exclude):
+        """Check whether intersections exist."""
+        intersections = mcs.get_intersections(include)
+        attr, intersections = ('__include__', intersections) \
+            if intersections \
+            else ('__exclude__', mcs.get_intersections(exclude))
+
+        if intersections:
+            raise AttributeError(
+                "It is not allowed to mention already defined properties: "
+                "{0} in {1} attributes.".format(", ".join(intersections),
+                                                attr))
+
+    @classmethod
+    def get_intersections(mcs, attr):
+        """Return intersection with defined properties if exists.
+
+        :type attr: list
+        :rtype: list
+        """
+        if not attr:
+            return []
+        return list(set(mcs.get_properties()).intersection(attr))
+
+    @classmethod
+    def match_unknown_attrs(mcs, include, exclude):
+        """Check about using nonexistent attributes."""
+        model_cls = mcs.attributes.get('__model_cls__')
+        unknown_attr = []
+        for item in include:
+            if not hasattr(model_cls, item):
+                unknown_attr.append(item)
+
+        for item in exclude:
+            if not hasattr(model_cls, item):
+                unknown_attr.append(item)
+
+        if unknown_attr:
+            raise AttributeError(
+                "Nonexistent attributes: {0}.".format(
+                    ", ".join(unknown_attr)))
+
+    @classmethod
+    def get_properties(mcs):
+        """Return list of names of defined properties.
+
+        :rtype: list
+        """
+        return [key for key, value in six.iteritems(mcs.attributes) if
+                isinstance(value, property)]
```

Comment:
Not sure you are right...
It is checking for a value is instance of property like this https://github.com/ets-labs/python-domain-models/pull/33/files#diff-469a50d37f12edde45a4bd1a8aaa762cR43
It is not an instance of models field.